packages/aws-cdk-lib/aws-ec2/test/security-group.test.ts

```diff
@@ -596,6 +596,30 @@ describe('security group lookup', () => {
 
   });
 
+  test('can look up a security group by filters', () => {
+    // GIVEN
+    const app = new App();
+    const stack = new Stack(app, 'stack', {
+      env: {
+        account: '1234',
+        region: 'us-east-1',
+      },
+    });
+
+    // WHEN
+    const securityGroup = SecurityGroup.fromLookupByFilters(stack, 'SG1', {
+      ownerId: '012345678901',
+      description: 'my description',
+      tagKeys: ['tagA', 'tagB'],
+      tags: { tagC: ['valueC', 'otherValueC'], tagD: ['valueD'] },
+    });
+
+    // THEN
+    expect(securityGroup.securityGroupId).toEqual('sg-12345678');
+    expect(securityGroup.allowAllOutbound).toEqual(true);
```

Comment:
what happens if this matches more than one security group ?

Reply:
An error gets thrown in `aws-cdk/lib/context-providers/security-group.ts` for that case (and a test already exists for that case)